core/dbt/parser/partial.py

```diff
@@ -903,6 +926,19 @@ def delete_schema_exposure(self, schema_file, exposure_dict):
             elif unique_id in self.saved_manifest.disabled:
                 self.delete_disabled(unique_id, schema_file.file_id)
 
+    # groups are created only from schema files, so just delete the group
```

Comment:
When there are references to groups in other nodes, then we'll have to go through and schedule every node with this groups for reparsing.

Reply:
Makes sense. That would also go for modifications to existing groups in addition to deletions, right? 

I'm thinking it makes more sense to add that to the follow-up PR that parses group configs on group-able nodes: https://github.com/dbt-labs/dbt-core/pull/6965.

Reply:
We'd only need to reparse for modifications of existing groups if information from the group is pulled into the node that needs to be rebuilt. If we only have the name in the node then probably not.

Reply:
On thinking about this... Would people be pulling group information into model jinja code? Like model.group.owner? Then we'd have to reparse.

Reply:
We're not storing any additional group information directly on the node, right? Users would need to look up `.groups()` in the manifest / `flat_graph` (a.k.a. `{{ graph }}` in Jinja code) to get information about the `owner`

Reply:
> We're not storing any additional group information directly on the node, right? Users would need to look up .groups() in the manifest / flat_graph (a.k.a. {{ graph }} in Jinja code) to get information about the owner

Right - we're just storing the name. Changes to which will be captured as a deleted + new group. We're not currently adding groups to the flat graph - should we be?

Reply:
Ooh, I'd just seen [this change](https://github.com/dbt-labs/dbt-core/pull/6921/files#diff-33c94e40283afdfeba411a98c0628fc610c287e41c3477d88559f1e0244259c3R421-R426) and thought we were.

I think we probably should. This feels like valid metadata that someone might want access to in the Jinja context (including the folks who maintain the [`dbt_artifacts`](https://github.com/brooklyn-data/dbt_artifacts) package), and I can't see potential for anti-pattern shenanigans. (There are indeed some things we exclude from the `flat_graph`, such as macros. I would say that exclusion is somewhat intentional. I just remembered this issue that I tagged myself to refine forever ago: https://github.com/dbt-labs/dbt-core/issues/4919)

It would just require an update to [this method](https://github.com/dbt-labs/dbt-core/blob/daea7d59a72da3a164d6e28839db8e240efd7688/core/dbt/contracts/graph/manifest.py#L679-L690), and associated tests

Reply:
okay, makes sense! I had actually added it on first pass of this work but wasn't convinced it should be part of the flat graph so removed it. Will add now